sysfs.c

```diff
@@ -179,6 +179,7 @@ struct mdinfo *sysfs_read(int fd, char *devnm, unsigned long options)
 			sra->array.major_version = -1;
 			sra->array.minor_version = -2;
 			strcpy(sra->text_version, buf+9);
+			sra->text_version[sizeof(sra->text_version) - 1] = '\0';
```

Comment:
why not?
```c
snprintf(sra->text_version, sizeof(sra->text_version), "/%s, buf + 9);
```

Reply:
because, it produces an error....

sysfs.c: In function ‘sysfs_read’:
sysfs.c:181:81: error: ‘%s’ directive output may be truncated writing up to 4086 bytes into a region of size 50 [-Werror=format-truncation=]
  181 |                    snprintf(sra->text_version, sizeof(sra->text_version), "%s", buf + 9);

Reply:
@mtkaczyk, this works...

-                       strcpy(sra->text_version, buf+9);
+                       char new_string[sizeof(sra->text_version)];
+                       strcpy(new_string, buf + 9);
+                       sprintf(sra->text_version, "%s", new_string);

Reply:
ok, this way is fine then :)